File: app/lib/services/api_service.dart
import 'dart:convert';
import 'package:http/http.dart' as http;
import '../config.dart';

class ApiService {
  static String get baseUrl => ServerConfig.apiBase;

  /// ===============================
  /// GET MAKANAN
  /// ===============================
  static Future<List<dynamic>> getMakanan() async {
    final response = await http.get(Uri.parse("$baseUrl/makanan"));

    if (response.statusCode == 200) {
      return jsonDecode(response.body);
    }

    throw Exception("Gagal mengambil data makanan");
  }

  /// ===============================
  /// LOGIN
  /// ===============================
  static Future<Map?> login(String email, String password) async {
    final response = await http.post(
      Uri.parse("$baseUrl/auth/login"),

      headers: {"Content-Type": "application/json"},

      body: jsonEncode({"email": email, "password": password}),
    );

    if (response.statusCode == 200) {
      return jsonDecode(response.body);
    }

    return null;
  }

  /// ===============================
  /// REGISTER
  /// ===============================
  static Future<Map?> register(
    String nama,
    String email,
    String password,
  ) async {
    final response = await http.post(
      Uri.parse("$baseUrl/auth/register"),

      headers: {"Content-Type": "application/json"},

      body: jsonEncode({"nama": nama, "email": email, "password": password}),
    );

    if (response.statusCode == 200) {
      return jsonDecode(response.body);
    }

    return null;
  }
}

File: frontend/src/services/api_service.js
/**
 * API Service — mirror dari Flutter api_service.dart
 *
 * Semua endpoint backend terpusat di sini.
 * Menggunakan ServerConfig untuk base URL.
 */

import ServerConfig from '../config';

const ApiService = {
  // ================================
  // GET MAKANAN
  // ================================
  async getMakanan() {
    const res = await fetch(`${ServerConfig.apiBase}/makanan`);

    if (!res.ok) throw new Error('Gagal mengambil data makanan');

    return res.json();
  },

  // ================================
  // GET DETAIL MAKANAN
  // ================================
  async getMakananById(id) {
    const res = await fetch(`${ServerConfig.apiBase}/makanan/${id}`);

    if (!res.ok) throw new Error('Gagal mengambil detail makanan');

    return res.json();
  },

  // ================================
  // LOGIN
  // ================================
  async login(email, password) {
    const res = await fetch(`${ServerConfig.apiBase}/auth/login`, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({ email, password }),
    });

    if (!res.ok) return null;

    return res.json();
  },

  // ================================
  // REGISTER
  // ================================
  async register(nama, email, password) {
    const res = await fetch(`${ServerConfig.apiBase}/auth/register`, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({ nama, email, password }),
    });

    if (!res.ok) return null;

    return res.json();
  },
};

export default ApiService;
